### ISSUE TITLE
"Credentials in the backup file are more recent" immediately after exporting credentials

### ISSUE THREAD

**Issue by ghalfacree** [mooltipass/moolticute#813]
Got an odd one here: When I save a new credential to Moolticute, it correctly prompts me to update my monitored backup file; when I do, it then immediately switches to telling me the backup file is more recent than the credentials on the device... despite the fact I literally *just* backed the device up into said backup file.

### Expected behavior

If you've just exported to the monitored backup file, by definition the backup file and on-device credentials should be a perfect match.

### Actual behavior

You receive a warning labelled "Import db backup" with the text "Credentials in the backup file are more recent. Do you want to import credentials to the device?" Oddly, if you press "No" or the cross on the dialogue it will immediately reappear... and keep doing so in a loop until, dozens of clicks later, Moolticute finally gives up asking.

### Step by step guide to reproduce the problem

1. Set up backup monitoring.
2. Save a new credential.
3. Accept Moolticute's offer to update the backup file.
4. The "import" dialogue should immediately appear once the backup is complete.

### Moolticute Version

v0.45.2

### Operating System

Ubuntu 20.04.2 LTS 64-bit

### Mooltipass Device

Mooltipass Mini BLE

**Comment by limpkin** [comment]
Hello!
Could you confirm this behavior with the 45-3 testign that was just released? :)

**Comment by ghalfacree** [comment]
Quit Moolticute, upgraded to 45-3-testing, reloaded Moolticute, unlocked the MBLE, immediately received an "import db backup" dialogue. This time, however, it went away after a single click on 'no' - so that's a definite improvement!

I then created a new credential in the Moolticute menu - site name 'example.net,' login 'test,' password 'test' - and saved it to the Mooltipass. The credential saved successfully... but I did not receive notification that the credentials on the MBLE are newer than the backup file.

Unplugged the MBLE and plugged it back in again, unlocked it, and I got a notification that credentials in the *backup file* are more recent - which is *not* the case, given I just created the test credential. If I click 'Yes,' the backup will be restored and I'll lose my latest credential - at least, that's my understanding: "Do Not Delete Credentials on Local DB" is unchecked by default, so I assume that also affects automatic backup restoration as well as a manual import?

Looks like something's screwy with the monitoring system, I'm afraid!

**Comment by ghalfacree** [comment]
Manually exported the credentials to the monitored file, and ended up with these two side-by-side...

![image](https://user-images.githubusercontent.com/1130041/108698625-40a3f680-74fc-11eb-939e-8ecdaebdecd8.png)

**Comment by ghalfacree** [comment]
I'm now running v0.45.4-testing, and it seems partially fixed: after updating a password, Moolticute correctly recognises that the credentials in the device are more recent than in the monitored backup file and prompts me to export the database.

But when I do, it exports fine... then brings up the same "credentials on the device are more recent" warning - despite the fact the credentials on the device and in the backup file should now be an exact match. If I allow it to export *again*, it seems to work the second time - the prompt goes away.

**Comment by deXol** [comment]
Hello @ghalfacree,
I am trying to reproduce the issue, but was not able so far:
![exportDB](https://user-images.githubusercontent.com/11043249/111376107-5772de80-869f-11eb-8e1b-1f27b5a1b793.gif)
After exporting the database from "Credentials on the device are more recent" prompt it is not displayed for me anymore.
Are you doing something differently than I did above? If not, can you please send the logs about the exports?

**Comment by ghalfacree** [comment]
@deXol Just tried to trigger it again, and failed - it correctly prompted once, then the prompt went away after the first export. Not sure why it would work fine this time and not yesterday - the only difference this time was I let the Mooltipass lock itself, so had to unlock it before export.

If it makes a difference, I'm running Moolticute on Linux with the monitored file on an ext4 partition mounted as `noatime` `nodiratime`.

**Comment by ghalfacree** [comment]
Okay, I've triggered it again - this time by adding a file to the MBLE.

What I did:

1. Updated a password.
2. Was *not* prompted to update the backup.
3. Disconnected, reconnected, unlocked the MBLE.
4. Was prompted to update the backup. Did so, worked fine - no spurious prompts.
5. Went to the Files tab, added a new file.
6. Immediately presented by a "Credentials in the backup file are more recent" dialogue.

I then cancelled the dialogue and triggered a manual export from the Synchronisation menu. As soon as that completed, a "Credentials on the device are more recent" warning appeared at the bottom of the Moolticute menu - which is definitely not the case, having literally just exported them.

I've captured a flow below. It demonstrates several stages of the bug: claiming the backup file is newer than the device when it isn't, claiming the same when a file has been added, and claiming the device is newer than the backup file literally immediately after manually exporting the database.

![Peek 2021-03-19 09-18](https://user-images.githubusercontent.com/1130041/111758378-56ef6900-8894-11eb-9536-d55408583875.gif)
